yt/visualization/plot_container.py

```diff
@@ -943,45 +990,26 @@ def set_zlim(self, field, zmin, zmax, dynamic_range=None):
 
         """
 
-        def _sanitize_units(z, _field):
-            # convert dimensionful inputs to float
-            if isinstance(z, tuple):
-                z = self.ds.quan(*z)
-            if isinstance(z, YTQuantity):
-                try:
-                    plot_units = self.frb[_field].units
-                    z = z.to(plot_units).value
-                except AttributeError:
-                    # only certain subclasses have a frb attribute
-                    # they can rely on for inspecting units
-                    mylog.warning(
-                        "%s class doesn't support zmin/zmax"
-                        " as tuples or unyt_quantitiy",
-                        self.__class__.__name__,
-                    )
-                    z = z.value
-            return z
-
         if field == "all":
             fields = list(self.plots.keys())
         else:
             fields = field
+        if zmin is None and zmax is None:
+            raise TypeError("Missing required argument zmin or zmax")
```

Comment:
Thank you so much for catching this early ! This is indeed a flaw in my current design, I would also _assume_ that setting either limit to `None` explicitly would restore the default, data limits.
Here I used `None` as a sentinel default (to signal "no value was passed"), so it doesn't work, but I can easily replace it with a custom sentinel value (say `Unset = object()`).
I'll start with adding a (broken) test and try to find a satisfactory solution keeping the rest of your comment in mind.

Reply:
I implemented my solution + a test
See the corresponding diff https://github.com/yt-project/yt/pull/3849/files/2876fd6fd02536194b4f08291f2ac83bdc981c89..b37f53785e0537f841053e768f9b5a7896238ad1

Reply:
mypy revealed that my solution isn't yet viable as it doesn't mix well with the `dynamic_range` parameter.
I will probably need to move the logic for it to the `NormHandler` class

Reply:
update: actually my implementation is still problematic because setting `p.set_zlim(field, zmin=None, dynamic_range=2)` doesn't work as you'd expect. On the other hand `p.set_zlim(field, zmin="min", dynamic_range=2)` does.
I'm going to push a temporary solution to fix this but I'm deprecating passing `None` explicitly here, and encouraging the use of `'min'` and `'max'` instead.

Reply:
@chrishavlin